CMakeLists.txt

```diff
@@ -849,6 +850,42 @@ IF(WITH_ORB_SLAM AND NOT G2O_FOUND)
     ENDIF(ORB_SLAM_FOUND)
 ENDIF(WITH_ORB_SLAM AND NOT G2O_FOUND)
 
+)))))))))))))))))))))))))))) TODO check this block below
+
+IF(WITH_CUVSLAM)
+    FIND_PACKAGE(cuvslam QUIET)
+    IF(cuvslam_FOUND)
+        MESSAGE(STATUS "Found cuVSLAM: ${cuvslam_INCLUDE_DIRS}")
+        SET(RTABMAP_CUVSLAM TRUE)
+    ELSE()
+        MESSAGE(STATUS "cuVSLAM not found, trying to find cuvslam headers...")
+        FIND_PATH(CUVSLAM_INCLUDE_DIR cuvslam/cuvslam.h
+            PATHS
+            /usr/include
+            /usr/local/include
+            /opt/cuvslam/include
+            $ENV{CUVSLAM_ROOT}/include
+            $ENV{CUVSLAM_ROOT_DIR}/include
+        )
+        FIND_LIBRARY(CUVSLAM_LIBRARY cuvslam
+            PATHS
+            /usr/lib
+            /usr/local/lib
+            /opt/cuvslam/lib
+            $ENV{CUVSLAM_ROOT}/lib
+            $ENV{CUVSLAM_ROOT_DIR}/lib
+        )
+        IF(CUVSLAM_INCLUDE_DIR AND CUVSLAM_LIBRARY)
+            SET(cuvslam_FOUND TRUE)
+            SET(cuvslam_INCLUDE_DIRS ${CUVSLAM_INCLUDE_DIR})
+            SET(cuvslam_LIBRARIES ${CUVSLAM_LIBRARY})
+            SET(RTABMAP_CUVSLAM TRUE)
+            MESSAGE(STATUS "Found cuVSLAM: ${cuvslam_INCLUDE_DIRS}")
+        ENDIF()
+    ENDIF()
+ENDIF(WITH_CUVSLAM)
+))))))))))))))))))))))))))))
+
```

Comment:
Could be moved into a `FindCuVSLAM.cmake` file under `cmake_modules` folder, if `cuvslam` library doesn't provide already a cmake config to find it.